trieste/acquisition/optimizer.py

```diff
@@ -162,16 +162,31 @@ def _objective() -> TensorType:
             return -target_func(bijector.forward(variable[:, None, :]))  # [1]
 
         chosen_points = tf.ones([0, len(space.lower)], dtype=trial_search_space.dtype)  # [0, D]
-        for i in tf.range(num_restarts):  # restart optimization
+        for i in tf.range(num_restarts):  # perfom optimization from chosen starting points
             variable.assign(bijector.inverse(initial_points[i : i + 1]))  # [1, D]
             gpflow.optimizers.Scipy().minimize(_objective, (variable,), **opt_kwargs)
             chosen_points = tf.concat(
                 [chosen_points, bijector.forward(variable)], axis=0
             )  # [i+1, D]
 
         chosen_func_values = target_func(chosen_points[:, None, :])  # [num_restarts, 1]
+        chosen_point = tf.gather(chosen_points, tf.argmax(chosen_func_values))  # [1, D]
```

Comment:
is this NaN-proof?

Reply:
weirdly it is. tf.argmax behaves like np.nanargmax